### PR TITLE
Contract Linting

### PR DESCRIPTION
**Pull request by mujahidkay**
Closes: https://github.com/Agoric/dapp-offer-up/issues/35

### Description
This PR enables typechecking via tsc and fixes lint errors

### CODE REVIEW

File: contract/package.json [hunk 1]
```diff
@@ -26,26 +26,29 @@
     "@endo/promise-kit": "^1.1.5",
     "@endo/ses-ava": "^1.2.5",
     "@jessie.js/eslint-plugin": "^0.4.1",
+    "@typescript-eslint/eslint-plugin": "^7.0.2",
+    "@typescript-eslint/parser": "^7.15.0",
```
Comment (turadg) [inline contract/package.json:30]: I believe the `typescript-eslint` dep suffices.
```suggestion
```

File: contract/test/test-build-proposal.js [hunk 2]
```diff
@@ -4,10 +4,31 @@ import { execSync } from 'node:child_process';
...
+  t.context = {
+    compressFile: makeCompressFile(fs.readFile),
+  };
+});
+
+/**
+ * @param {ExecutionContext<TestContext>} t
+ */
 
 test.only('proposal builder generates compressed bundles less than 1MB', async t => {
```
Comment (turadg) [inline contract/test/test-build-proposal.js:28]: drive-by, but please take `.only` out of the above test. It's already the only test in this module.

File: contract/test/test-build-proposal.js [hunk 3]
```diff
@@ -17,7 +38,9 @@ test.only('proposal builder generates compressed bundles less than 1MB', async t
 
   for (const bundle of bundles) {
     // eslint-disable-next-line @jessie.js/safe-await-separator
-    const buffer = await t.context.compressFile(bundle);
+    const buffer = await /** @type {TestContext} */ (t.context).compressFile(
```
Comment (turadg) [inline contract/test/test-build-proposal.js:41]: `t.context` type should be implied by the type of `t`. What happens without this cast?

File: contract/test/test-build-proposal.js [hunk 4]
```diff
@@ -4,10 +4,31 @@ import { execSync } from 'node:child_process';
...
+ * @import {ExecutionContext} from 'ava';
+ */
+/**
+ * @typedef {object} TestContext
+ * @property {function(any): Promise<Buffer>} compressFile
+ */
+
+/**
+ * @param {ExecutionContext<TestContext>} t
+ */
```
Comment (turadg) [inline contract/test/test-build-proposal.js:17]: please just type `test`. the rest will fall out of that.
https://github.com/Agoric/agoric-sdk/blob/136f1caa919428684a88f2213d0cd9b3185f8a33/packages/boot/test/configs.test.js#L20-L24

```js
const test =
  /** @type {import('ava').TestFn<Awaited<ReturnType<typeof makeTestContext>>>}} */ (
    anyTest
  );
```

File: contract/test/test-contract.js [hunk 5]
```diff
@@ -22,19 +23,33 @@ import { startOfferUpContract } from '../src/offer-up-proposal.js';
 const myRequire = createRequire(import.meta.url);
 const contractPath = myRequire.resolve(`../src/offer-up.contract.js`);
 
-/** @type {import('ava').TestFn<Awaited<ReturnType<makeTestContext>>>} */
-const test = anyTest;
```
Comment (turadg) [inline contract/test/test-contract.js:26]: why change this?

File: contract/test/test-build-proposal.js (hunk 3 above)
Comment (mujahidkay) [reply contract/test/test-build-proposal.js:41]: It's implied to be `undefined`. And says `t.context is of type unknown`

`/ava/types/test-fn.d.cts`
```js
/** The `t` value passed to test & hook implementations. */
export type ExecutionContext<Context = unknown> = {
	/** Test context, shared with hooks. */
	context: Context;

	/** Title of the test or hook. */
	readonly title: string;
	...
}
```

Comment (turadg) [reply contract/test/test-build-proposal.js]: That's because `test` wasn't typed. See 064eb83

Comment (mujahidkay) [reply contract/test/test-build-proposal.js]: neat. thanks for this! 🚀

File: contract/test/test-contract.js (hunk 5 above)
Comment (mujahidkay) [reply contract/test/test-contract.js:26]: sorry, wasn't sure what this was achieving but I now finally understand. reverting while addressing comments.

Comment (turadg) [reply contract/test/test-contract.js:26]: reminds me of https://thoughtbot.com/blog/chestertons-fence =)

File: contract/test/test-contract.js [hunk 6]
```diff
@@ -59,7 +69,7 @@ test('Install the contract', async t => {
 });
 
 test('Start the contract', async t => {
-  const { zoe, bundle } = t.context;
+  const { zoe, bundle } = /** @type {TestContext} */ (t.context);
```
Comment (turadg) [inline contract/test/test-contract.js:72]: now that `test` is typed, all these casts of `t.context` are redundant.

please remove

**Review by turadg — APPROVED** [review]
Trusting you'll remove the `t.context` casts
